sklearn/metrics/pairwise.py

```diff
@@ -1985,16 +2002,29 @@ def _pairwise_callable(X, Y, metric, ensure_all_finite=True, **kwds):
         # No input dimension checking done for custom metrics (left to user)
         ensure_2d=False,
     )
+    _, _, dtype_float = _find_floating_dtype_allow_sparse(X, Y, xp=xp)
+
+    def _get_slice(array, index):
+        # TODO : Support for 1D shapes in scipy sparse arrays (COO, DOK and CSR
+        # formats) only added in 1.14. We must return 2D array until min scipy 1.14
+        # i.e. below 2 lines can be removed once min scipy >= 1.14
+        if issparse(array):
+            return array[[index], :]
+        # When `metric` is a callable, 1D input arrays allowed, in which case
+        # scalar should be returned.
+        if array.ndim == 1:
+            return array[index]
+        else:
+            return array[index, ...]
```

Comment:
I don't love that I've had to add this but I think this is the only option until we drop older scipy.
Note there are 3 different slice types and it would have been too confusing as a one liner.